suite/tests/linux/syscall_pwait.c

```diff
@@ -134,24 +149,148 @@ main(int argc, char *argv[])
     if (pid < 0) {
         perror("fork error");
     } else if (pid == 0) {
-        /* waste some time */
-        nanosleep(&sleeptime, NULL);
-        kill(getppid(), SIGUSR2);
-        /* waste some time */
-        nanosleep(&sleeptime, NULL);
-        kill(getppid(), SIGUSR1);
-        /* waste some time */
-        nanosleep(&sleeptime, NULL);
-        kill(getppid(), SIGUSR1);
-        return 0;
+        return kick_off_child_signals(&sleeptime);
     }
 
+    print("Testing ppoll\n");
+
     count = 0;
     while (count++ < 3) {
         /* XXX i#3240: DR currently does not handle the atomicity aspect of this system
-         * call. Once it does, please include this in this test or add a new test. */
-        ppoll(NULL, 0, NULL, &test_set);
+         * call. Once it does, please include this in this test or add a new test.
+         */
+        if (ppoll(NULL, 0, NULL, &test_set) == -1) {
+            if (errno != EINTR)
+                perror("expected EINTR");
+        } else {
+            perror("expected interruption of syscall");
+        }
```

Comment:
optional: There's still a lot of repetition: you could imagine having a single function that tests one of these syscalls and it just takes in the actual glibc call, as the rest is all identical.

Reply:
I've re-factored a little bit more, some duplications still left but the parameters are all different and it's easy to read.